packages/vats/src/orchestration.js

```diff
@@ -0,0 +1,255 @@
+     *   localAddr: string,
+     *   remoteAddr: string,
+     * ) => void} [onOpen]
+     *   optional cb handler for connection creation
+     * @param {(connection: Connection, reason: unknown) => void} [onClose]
+     *   optional cb handler for connection closing
+     * @param {(
+     *   connection: Connection,
+     *   bytes: Uint8Array,
+     * ) => PromiseVow<string>} [onReceive]
+     *   optional cb handler for connection recieving packet
+     */
+    (onOpen, onClose, onReceive) => ({ onOpen, onClose, onReceive }),
+    {
+      async onOpen(connection, localAddr, remoteAddr) {
+        trace(`ICA Channel Opened for ${localAddr} at ${remoteAddr}`);
+        const { onOpen } = this.state;
+        if (onOpen && typeof onOpen === 'function') {
+          void onOpen(connection, localAddr, remoteAddr);
+        }
+      },
+      async onClose(connection, reason) {
+        trace(`ICA Channel closed. Reason: ${reason}`);
+        const { onClose } = this.state;
+        if (onClose && typeof onClose === 'function') {
+          void onClose(connection, reason);
+        }
+      },
+      async onReceive(connection, bytes) {
+        const { onReceive } = this.state;
+        if (onReceive && typeof onReceive === 'function') {
+          void onReceive(connection, bytes);
+        }
+        return '';
+      },
+    },
+  );
+
+export const OrchestrationI = M.interface('Orchestration', {
+  provideAccount: M.callWhen(M.string(), M.string())
+    .optional(M.remotable('Port'))
+    .returns(M.remotable('ChainAccount')),
+  getChain: M.callWhen(M.string()).returns(M.remotable('Chain')),
+});
+
+/**
+ * @param {import('@agoric/base-zone').Zone} zone
+ * @param {ReturnType<typeof prepareChainAccount>} createChainAccount
+ * @param {ReturnType<typeof prepareConnectionHandler>} makeConnectionHandler
+ */
+const prepareOrchestration = (
+  zone,
+  createChainAccount,
+  makeConnectionHandler,
+) =>
+  zone.exoClassKit(
+    'Orchestration',
+    {
+      admin: M.interface('OrchestrationAdmin', {
+        foo: M.call().returns(M.string()),
+      }),
+      public: OrchestrationI,
+    },
+    /** @param {Partial<OrchestrationPowers>} [initialPowers] */
+    initialPowers => {
+      /** @type {PowerStore} */
+      const powers = zone.detached().mapStore('PowerStore');
+      if (initialPowers) {
+        for (const [name, power] of Object.entries(initialPowers)) {
+          powers.init(/** @type {keyof OrchestrationPowers} */ (name), power);
+        }
+      }
+      return { powers, icaControllerNonce: 0 };
+    },
+    {
+      admin: {
+        foo() {
+          return 'bar';
+        },
+      },
+      public: {
+        /**
+         * @param {string} hostConnectionId the counterparty connection_id
+         * @param {string} controllerConnectionId self connection_id
+         * @param {Port} [currentPort] if a port is provided, it will be reused
+         *   to create the account
+         */
+        async provideAccount(
```

Comment:
The `provide` prefix means find-or-create; i.e. if it's already been created, return the existing one. This code makes a fresh account on every call.

```suggestion
        async makeAccount(
```